clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslServerAuthenticator.java

```diff
@@ -433,7 +437,7 @@ private void handleSaslToken(byte[] clientToken) throws IOException {
             RequestHeader header = RequestHeader.parse(requestBuffer);
             ApiKeys apiKey = header.apiKey();
             short version = header.apiVersion();
-            RequestContext requestContext = new RequestContext(header, connectionId, clientAddress(),
+            RequestContext requestContext = new RequestContext(header, connectionId, clientAddress(), Optional.of(clientPort()),
```

Comment:
I'm not quite sure the optionality is right here. Shouldn't there be the option of `Optional.empty()` if there's no data?

Reply:
Hmmm, the reason I kept clientPort Optional in RequestContext is because the data is not required in case of forwarded requests but wherever it can be determined by explicit value then that has been wrapped. I am not of an opinion to make a decision of returning Optional.empty in case connection is closed or socket exception occurred as that scenario ll anyways not make the channel usable.